src/commands.js

```diff
@@ -1,59 +1,29 @@
 const sleep = require('sleep');
-
+const {TextChannel, GuildChanne} = require('discord.js')
 //Clears the chat logs
-function clearMessages(message)
-{
-  //Gets all messages in the bot-commands channel
-  message.channel.fetchMessages().then(function(messages)
-  {
-    //Bulk deletes all the messages
-    message.channel.bulkDelete(messages);
-  });
-
-  //Sends a message
-  message.channel.send("Chat log has been cleared!").then(function(thisMessage)
-  {
-    //After the message it sent, it waits three seconds and deletes itself
-    sleep.sleep(3);
-    thisMessage.delete();
-  });
+function bulkDelete(textchan) {
+	if (textchan instanceof TextChannel && textchan.guild.me.hasPermission("MANAGE_MESSAGES") || textchan.memberPermissions(textchan.guild.me).has("MANAGE_MESSAGES")) {
+		return true
+		//Gets all messages in the bot-commands channel
+		textchan.fetchMessages().then(function(messages) {
+		//Bulk deletes all the messages
+		textchan.bulkDelete(messages);
+	});
+	} else {
+		return null
+	}
 }
-
-function linkSubreddit(message) {
-  //String of the message
-  var messageStr = message.toString();
-
-  //Gets starting index of the subreddit
-  var subCutoff = messageStr.substring(messageStr.indexOf('r/'), messageStr.length);
-
-  //Check for a space to just get the raw subreddit
-  var endingIndex = (subCutoff.indexOf(' ') > -1) ? subCutoff.indexOf(' ') : subCutoff.length;
-
-  //Substring for the actual subreddit
-  var subredditStr = subCutoff.substring(0, endingIndex);
-
-  message.channel.send("OwO, what's this? I see a mention of a subreddit! Here's a link: http://reddit.com/"+subredditStr, {reply: message.author});
+function subReddit(message) {
```

Comment:
So did you put both of my subreddit functions into just one?

Reply:
I see, I was planning on using something else with that later, but this works just fine